common/recordCreate.js

```diff
@@ -0,0 +1,308 @@
+(function() {
+    'use strict';
+    angular.module('chaise.recordcreate', ['chaise.errors','chaise.utils']).factory("recordCreate", ['$cookies', '$log', '$window', '$uibModal', 'AlertsService', 'DataUtils', 'UriUtils', 'modalBox',
+     function($cookies, $log, $window, $uibModal, AlertsService, DataUtils, UriUtils, modalBox) {
+
+        var viewModel = {};
+        var GV_recordEditModel = {},
+            completed = {};
+        var addRecordRequests = {}; // <generated unique id : reference of related table>
+        var editRecordRequests = {}; // generated id: {schemaName, tableName}
+        var updated = {};
+
+        /**
+         * checkUpdate - to check all recrds are updated; passed as callback to uploadFiles().
+         *
+         * @param  {array} submissionRowsCopy   array contains updated recrds attributes
+         * @param  {object} rootScope           object contains global values from calling function
+         * @return  {array} data                key-value pair of updated records
+         */
+        function checkUpdate(submissionRowsCopy, rootScope){
+            /**
+             * After uploading files, the returned submissionRowsCopy contains
+             * new file data. This includes filename, filebyte, and md5.
+             * The following makes sure that all the data are updated.
+             * That's why this for loop must be after uploading files and not before.
+             * And we cannot just pass submissionRowsCopy to update function, because
+             * update function only accepts array of tuples (and not just key-value pair).
+             */
+            for (var i = 0; i < submissionRowsCopy.length; i++) {
+                var row = submissionRowsCopy[i];
+                var data = rootScope.tuples[i].data;
+                // assign each value from the form to the data object on tuple
+                for (var key in row) {
+                    data[key] = (row[key] === '' ? null : row[key]);
+                }
+            }
+            return data;
+        }
+
+        /**
+         * uploadFiles - uploading files
+         *
+         * @param  {array} submissionRowsCopy   data that is going to be uploaded
+         * @param  {object} rootScope           object contains global values from calling function
+         * @param  {object} onSuccess           callback
+         */
+        function uploadFiles(submissionRowsCopy, rootScope, onSuccess) {
```

Comment:
Let me be more explicit, the reference should be passed into this function not rootscope. We should not pass the entire rootscope into this function, there are far more functions and properties on the rootscope that are not needed.